File: src/two_sum.py
#!/usr/bin/env python3
################################################################################
#
#   Filename:           two_sum.py
#
#   Author:             Arnaud Ongenae
#
#   Leetcode.com:       problem #1
#
#   Problem description:
#   https://leetcode.com/problems/two-sum/description/
#
################################################################################


class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        buffer = {}
        for index, value in enumerate(nums):
            if value in buffer:
                return [buffer[value], index]
            else:
                buffer[target-value] = index

File: src/two_sum.go
/*******************************************************************************
*
*   Filename:           two_sum.go
*
*   Author:             Arnaud Ongenae
*
*   Leetcode.com:       problem #1
*
*   Problem description:
*   https://leetcode.com/problems/two-sum/description/
*
*******************************************************************************/

package main

func twoSum(nums []int, target int) []int {
	buffer := make(map[int]int)
	for index, value := range nums {
		if stored_index, ok := buffer[value]; ok {
			return []int{stored_index, index}
		} else {
			buffer[target-value] = index
		}
	}
	return []int{} // should not reach this line if solution exists
}
